editor/editor_help_search.cpp

```diff
@@ -590,19 +657,49 @@ TreeItem *EditorHelpSearch::Runner::_create_class_hierarchy(const ClassMatch &p_
 	return class_item;
 }
 
+TreeItem *EditorHelpSearch::Runner::_find_or_create_item(TreeItem *p_parent, const String &p_item_meta) {
+	TreeItem *item = nullptr;
+
+	// Attempt to find in cache.
+	if (tree_cache->item_cache.has(p_item_meta)) {
+		item = tree_cache->item_cache[p_item_meta];
+
+		// Remove from cache.
+		tree_cache->item_cache.erase(p_item_meta);
+
+		// Add to tree.
+		p_parent->add_child(item);
+
+		// Erase buttons if any, to update icons.
+		for (int i = 0; i < item->get_button_count(0); ++i) {
+			item->erase_button(0, i);
+		}
+	} else {
+		// Otherwise create item.
+		item = results_tree->create_item(p_parent);
+	}
+
+	return item;
+}
+
 TreeItem *EditorHelpSearch::Runner::_create_class_item(TreeItem *p_parent, const DocData::ClassDoc *p_doc, bool p_gray) {
 	String tooltip = DTR(p_doc->brief_description.strip_edges());
 
-	TreeItem *item = results_tree->create_item(p_parent);
+	const String item_meta = "class_name:" + p_doc->name;
+
+	TreeItem *item = _find_or_create_item(p_parent, item_meta);
 	item->set_icon(0, EditorNode::get_singleton()->get_class_icon(p_doc->name));
 	item->set_text(0, p_doc->name);
 	item->set_text(1, TTR("Class"));
 	item->set_tooltip_text(0, tooltip);
 	item->set_tooltip_text(1, tooltip);
-	item->set_metadata(0, "class_name:" + p_doc->name);
+	item->set_metadata(0, item_meta);
 	if (p_gray) {
 		item->set_custom_color(0, disabled_color);
 		item->set_custom_color(1, disabled_color);
+	} else {
```

Comment:
Had forgotten to add this part to this stage, which would cause the disabled state to linger, now fixed